creator-node/src/diskManager.js

```diff
@@ -0,0 +1,107 @@
+const path = require('path')
+const fs = require('fs')
+const config = require('./config')
+
+// use this set to cache existing directory paths we know we've created so we don't make extraneous file system calls
+let EXISTING_PATHS = new Set()
+
+// regex to check if a directory or just a regular file
+// if directory - will have both outer and inner properties in match.groups
+// else - will have just outer property, no inner
+const CID_DIRECTORY_REGEX = /\/(?<outer>Qm[a-zA-Z0-9]{44})\/?(?<inner>Qm[a-zA-Z0-9]{44})?/
+
+class DiskManager {
+  /**
+   * Return the storagePath from the config
+   */
+  static getConfigStoragePath () {
+    return config.get('storagePath')
+  }
+
+  /**
+   * Construct the path to a file or directory
+   *
+   * eg. if you have a file `Qmabcxyz`, use this function to get the path /file_storage/files/cxy/Qmabcxyz
+   * eg. if you have a dir `Qmdir123`, use this function to get the path /file_storage/files/r12/Qmdir123/
+   * Use `computeFilePathInDir` if you want to get the path for a file inside a directory.
+   *
+   * @dev Returns a path with the three characters before the last character
+   *      eg QmYfSQCgCwhxwYcdEwCkFJHicDe6rzCAb7AtLz3GrHmuU6 will be eg /file_storage/muU/QmYfSQCgCwhxwYcdEwCkFJHicDe6rzCAb7AtLz3GrHmuU6
+   * @param {String} fsDest file system destination, either filename or directory name
+   */
+  static computeBasePath (fsDest) {
+    if (!fsDest || fsDest.length < 4) throw new Error(`Please pass in a valid fsDest to computeBasePath. Passed in ${fsDest}`)
+    if (fsDest.includes('/')) throw new Error('Cannot pass in a directory path into this function, please pass in the leaf dir or file name')
```

Comment:
if this is guaranteed to always be a CID, can we improve this validatio by checking that format explicitly? i think theres helpers somewhere

Reply:
not a blocker but if easy to do might be worth it

Reply:
it's not always a CID (eg the case of a the original track artifacts uploaded it's a folder with a UUID name). also i don't want to add extra validation in the event we want to generalize this function because that may impact backwards compatibility

Reply:
makes sense agreed

Reply:
oh i had a similar comment about this

Reply:
so this doesn't accept uuid's anymore. but should we do CID validation, i'm honestly not convinced one way or the other